analysis.go

```diff
@@ -102,55 +106,57 @@ func checkProjectTask(checkFactory getProjectCheckFactory, project, basepath str
 	}
 }
 
-// getDumpedResourceAsStruct retrieves the requested resource from the dump directory and
-// decodes the JSON into the provided interface
-func getDumpedResource(path []string, dest interface{}) error {
-	filePath := filepath.Join(path...)
-	contents, err := ioutil.ReadFile(filePath)
+// Takes an array of strings describing the path to a JSON file
+// which will be parsed and loaded into the supplied interface
+type DumpedJSONResourceFactory func([]string, interface{}) error
+
+// Returns a factory which will load resource from a given basepath. The factory parses the file
+// contents as JSON and loads it into the provided dest interface
+func getDumpedJSONResourceFactory(basepath []string) DumpedJSONResourceFactory {
+	return func(path []string, dest interface{}) error {
+		file := filepath.Join(append(basepath, path...)...)
+		contents, err := ioutil.ReadFile(file)
+
+		decoder := json.NewDecoder(bytes.NewBuffer(contents))
+		err = decoder.Decode(&dest)
+		if err != nil {
+			return err
+		}
 
-	decoder := json.NewDecoder(bytes.NewBuffer(contents))
-	err = decoder.Decode(&dest)
-	if err != nil {
-		return err
+		return nil
 	}
-
-	return nil
 }
 
-// CheckImagePullBackOff checks all events in the supplied project and if any
-// are exhibiting signs that they have experienced an ImagePullBackOff recently
-// this will be reflected in the returned Result data. Any errors are written to
-// the supplied stdErr writer.
-func CheckImagePullBackOff(project, basepath string) (Result, error) {
-	result := Result{Status: 0, StatusMessage: "this issue was not detected", CheckName: "check deploys for ImagePullBackOff error"}
+// CheckEventLogForErrors checks all events in the supplied project and if any
+// are not type 'Normal' (i.e. Warning or Error), it will add them to the returned results.
+func CheckEventLogForErrors(JSONResourceFactory DumpedJSONResourceFactory) (Result, error) {
+	result := Result{Status: 0, StatusMessage: "this issue was not detected", CheckName: "check eventlog for any errors"}
 	events := Events{}
-	err := getDumpedResource([]string{basepath, "projects", project, "definitions", "events.json"}, &events)
+	err := JSONResourceFactory([]string{"definitions", "events.json"}, &events)
 	if err != nil {
 		result.Status = 2
 		result.StatusMessage = "Error executing task"
 		return result, err
 	}
 
 	for _, event := range events.Items {
-		if event.Reason == "FailedSync" && strings.Contains(event.Message, "ImagePullBackOff") {
-			info := Info{Name: event.InvolvedObject.Name, Namespace: event.InvolvedObject.Namespace, Kind: event.Kind, Count: event.Count, Message: event.Message}
+		if event.Type != "Normal" {
```

Comment:
The Types are the likes of: 'Normal', 'warning', etc... Reasons are more like 'FailedSync' or 'DeploymentCreated', etc.